integreat_cms/cms/utils/repair_tree.py

```diff
@@ -0,0 +1,284 @@
+        for tree_node in mptt_fixer.get_fixed_tree_of_page(root_node.pk):
+            print_changed_fields(
+                Page.objects.get(id=tree_node.pk), tree_node.lft, tree_node.rgt, printer
+            )
+
+    if commit:
+        for page in mptt_fixer.get_fixed_tree_nodes():
+            page.save()
+
+
+def print_changed_fields(
+    tree_node: Page, left: int, right: int, printer: Printer = Printer()
+) -> None:
+    """
+    Check whether the tree fields are correct
+    """
+    printer.write(printer.bold(f"Page {tree_node.id}:"))
+    printer.success("\tparent_id: %i", tree_node.parent_id)
+    if not tree_node.parent or tree_node.tree_id == tree_node.parent.tree_id:
+        printer.success("\ttree_id: {tree_node.tree_id}")
+    else:
+        printer.error("\ttree_id: %i → %i", tree_node.tree_id, tree_node.parent.tree_id)
+    if tree_node.parent_id:
+        if tree_node.depth == tree_node.parent.depth + 1:
+            printer.success("\tdepth: %i", tree_node.depth)
+        else:
+            printer.error(
+                "\tdepth: %i → %i", tree_node.depth, tree_node.parent.depth + 1
+            )
+    elif tree_node.depth == 1:
+        printer.success("\tdepth: %i", tree_node.depth)
+    else:
+        printer.error("\tdepth: %i → 1", tree_node.depth)
+    if tree_node.lft == left:
+        printer.success("\tlft: %i", tree_node.lft)
+    else:
+        printer.error("\tlft: %i → %i", tree_node.lft, left)
+    if tree_node.rgt == right:
+        printer.success("\trgt: %i", tree_node.rgt)
+    else:
+        printer.error("\trgt: %i → %i", tree_node.rgt, right)
+
+
+class MPTTFixer:
+    """
+    Gets ALL nodes and coughs out fixed LFT, RGT and depth values. Uses the parent field
+    to fix hierarchy and sorts siblings by (potentially inconsistent) lft.
+    """
+
+    def __init__(self) -> None:
+        """
+        Creates a fixed tree when initializing class but does not save results
+        """
+        self.broken_nodes: list[Page] = list(
+            Page.objects.all().order_by("tree_id", "lft")
+        )
+        self.fixed_nodes: dict[int, Page] = {}
+        self.fix_root_nodes()
+        self.fix_child_nodes()
+
+    def fix_root_nodes(self) -> None:
+        """
+        extract root nodes and reset lft + rgt values
+        """
+        tree_counter = 1
+        for node in self.broken_nodes:
+            if not node.parent_id:
+                node.lft = 1
+                node.rgt = 2
+                node.depth = 1
+                node.fixed_children = []
+                node.tree_id = tree_counter
+                tree_counter = tree_counter + 1
+                self.fixed_nodes[node.pk] = node
+
+    def fix_child_nodes(self) -> None:
+        """
+        Get all remaining (child) nodes, add add them to the new/fixed tree
+        """
+        for node in self.broken_nodes:
+            if node.parent_id:
+                parent = self.fixed_nodes[node.parent_id]
```

Comment:
If the tree structure is broken, we cannot assume that the parent's `lft` value is necessarily smaller than the node's, so this might throw a `KeyError` if the parent is not yet fixed.